app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php

```diff
@@ -150,6 +150,16 @@ public function isShoppingCartPage()
             && (Mage::app()->getRequest()->getControllerName() === 'cart');
     }
 
+    /**
+     * @return bool
+     */
+    public function isProductPageCheckout()
+    {
+        $checkoutSession = Mage::getSingleton('checkout/session');
+
+        return $checkoutSession->getData('checkout_type') == Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE ? true : false;
```

Comment:
This could be simplified as `return $checkoutSession->getData('checkout_type') == Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE;`

Reply:
Thank @matt-thomason, I've updated it.